src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1073,7 +1073,14 @@ private static void ReportImplicitImplementationMatchDiagnostics(Symbol interfac
             if (implicitImpl.ContainsTupleNames() && MemberSignatureComparer.ConsideringTupleNamesCreatesDifference(implicitImpl, interfaceMember))
             {
                 // it is ok to implement implicitly with no tuple names, for compatibility with C# 6, but otherwise names should match
-                diagnostics.Add(ErrorCode.ERR_ImplBadTupleNames, implicitImpl.Locations[0], implicitImpl, interfaceMember);
+                if (implicitImpl.ContainingType == implementingType)
```

Comment:
>if (implicitImpl.ContainingType == implementingType) [](start = 16, length = 52)

Rather than adding a branching that pretty much duplicates the code with some minor variation, consider adding a local function that will return the right location based on conditional logic inside it.

Reply:
Or use a conditional operator for the `location` argument in `Add`.